### PR TITLE
finegrained replLock and making record channel send context aware

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

File: flow/model/cdc_stream.go [hunk 1]
```diff
@@ -39,8 +43,19 @@ func (r *CDCStream[T]) GetLastCheckpoint() int64 {
...
-func (r *CDCStream[T]) AddRecord(record Record[T]) {
-	r.records <- record
+func (r *CDCStream[T]) AddRecord(ctx context.Context, record Record[T]) {
+	logger := logger.LoggerFromCtx(ctx)
+	for {
+		select {
+		case r.records <- record:
+			return
+		case <-time.After(10 * time.Second):
+			logger.Warn("waiting on adding record to stream", slog.Any("record", record))
```
Comment (serprex) [inline flow/model/cdc_stream.go:53]:
```suggestion
```
Adding check on context cancel is fair as receiver may leave channel blocked, but no need for time.After

File: flow/model/cdc_stream.go [hunk 2]
```diff
@@ -39,8 +43,19 @@ func (r *CDCStream[T]) GetLastCheckpoint() int64 {
...
+		case <-ctx.Done():
+			logger.Warn("context cancelled while adding record to stream", slog.Any("record", record))
```
Comment (serprex) [inline flow/model/cdc_stream.go:55]: return error

File: flow/model/cdc_stream.go (hunk 1 above)
Comment (heavycrystal) [reply flow/model/cdc_stream.go:53]: want it for logging purposes mainly, to debug future issues

will change to timer

File: flow/connectors/postgres/cdc.go [hunk 3]
```diff
@@ -334,7 +346,10 @@ func PullCdcRecords[Items model.Items](
 		if err != nil {
 			return err
 		}
-		records.AddRecord(rec)
+		err = records.AddRecord(ctx, rec)
+		if err != nil {
+			return err
+		}
```
Comment (serprex) [inline flow/connectors/postgres/cdc.go:352]:
```suggestion
		if err := records.AddRecord(ctx, rec); err != nil {
			return err
		}
```

**Review by serprex — APPROVED** [review]

**Pull request merged**